examples/tree_2d_dgsem/elixir_mhdmultiion_collisions.jl

```diff
@@ -0,0 +1,151 @@
+# Characteristic velocity: V_inf = 1.00E+06 m/s
+#
+# The results of the paper can be reproduced using `source_terms = source_terms_collision_ion_ion` (i.e., only
+# taking into account ion-ion collisions). However, we include ion-electron collisions assuming a constant 
+# electron temperature of 1 keV in this elixir to test the function `source_terms_collision_ion_electron`
+
+# Return the electron pressure for a constant electron temperature Te = 1 keV
+function electron_pressure_constantTe(u, equations::IdealGlmMhdMultiIonEquations2D)
+    @unpack charge_to_mass = equations
+    Te = 0.008029953773 # [nondimensional] = 1 [keV]
+    total_electron_charge = zero(eltype(u))
+    for k in eachcomponent(equations)
+        rho_k = u[3 + (k - 1) * 5 + 1]
+        total_electron_charge += rho_k * charge_to_mass[k]
+    end
+
+    # Boltzmann constant divided by elementary charge
+    kB_e = 7.86319034E-02 #[nondimensional]
+
+    return total_electron_charge * kB_e * Te
+end
+
+# Return the constant electron temperature Te = 1 keV
+function electron_temperature_constantTe(u, equations::IdealGlmMhdMultiIonEquations2D)
+    return 0.008029953773 # [nondimensional] = 1 [keV]
+end
+
+# semidiscretization of the ideal MHD equations
+equations = IdealGlmMhdMultiIonEquations2D(gammas = (5 / 3, 5 / 3),
+                                           charge_to_mass = (76.3049060157692000,
+                                                             76.3049060157692000), # [nondimensional]
+                                           gas_constants = (1.0, 1.0), # [nondimensional]
+                                           molar_masses = (1.0, 1.0), # [nondimensional]
+                                           ion_ion_collision_constants = [0.0 0.4079382480442680;
+                                                                          0.4079382480442680 0.0], # [nondimensional] (computed with eq (4.142) of Schunk & Nagy (2009))
+                                           ion_electron_collision_constants = (8.56368379833E-06,
+                                                                               8.56368379833E-06), # [nondimensional] (computed with eq (9) of Ghosh et al. (2019))
+                                           electron_pressure = electron_pressure_constantTe,
+                                           electron_temperature = electron_temperature_constantTe,
+                                           initial_c_h = 0.0) # Deactivate GLM divergence cleaning
+
+# Frictional slowing of an ionized carbon fluid with respect to another background carbon fluid in motion
+function initial_condition_slow_down(x, t, equations::IdealGlmMhdMultiIonEquations2D)
+    v11 = 0.65508770000000
+    v21 = 0.0
+    v2 = v3 = 0.0
+    B1 = B2 = B3 = 0.0
+    rho1 = 0.1
+    rho2 = 1.0
+
+    p1 = 0.00040170535986
+    p2 = 0.00401705359856
+
+    return prim2cons(SVector(B1, B2, B3, rho1, v11, v2, v3, p1, rho2, v21, v2, v3, p2, 0.0),
```

Comment:
maybe initialize `0.0` as `0`, `1.0` as `1` and explicitly mention `psi = 0.0` or `psi= 0`?

Reply:
Is there a convention for the elixirs in the `examples` folder?... I thought everything in `src` had to be single-precision compatible, but not necessarily in `examples`(?).. In fact, a lot of other elixirs use combinations of number representations: 1.0, 2, 0.61, etc.

Reply:
If reasonably possible, we should make the examples compatible with other floating point types as well. If it's too much overhead, `Float64` is fine.

Reply:
Ok, I made the initial condition and helper functions of this example compatible with other floating point types in 28235a664df300f86d9620884ea48b4665a061c7. 

In my opinion, however, it makes it a bit less readable and, in this particular physically relevant example, it does not make any difference as the physical quantities (charge-to-mass ratios, collision frequencies, etc.) are provided in `Float64`.

Reply:
> explicitly mention psi = 0.0 or psi= 0?

Done in c8b4bf4417c9d3590551d57722fe743f5858d39c.